gamemodes/terrortown/entities/entities/ttt_hat_deerstalker.lua

```diff
@@ -21,25 +21,27 @@ end
 ---
 -- @realm shared
 function ENT:Initialize()
-	self:SetBeingWorn(true)
-
 	self:SetModel(self.Model)
 
 	self:DrawShadow(false)
 
 	-- don't physicsinit the ent here, because physicsing and then setting
 	-- movetype none is 1) a waste of memory, 2) broken
 
-	self:SetMoveType(MOVETYPE_NONE)
-	self:SetSolid(SOLID_NONE)
-	self:SetCollisionGroup(COLLISION_GROUP_DEBRIS)
-
 	if SERVER then
-		self.Wearer = self:GetParent()
+		if IsValid(self:GetParent()) then
+			self:EquipTo(self:GetParent())
+		else
+			self:Drop()
```

Comment:
Did this happen in your tests?
Seems weird in the Initialize logic.

Reply:
In the original code, it assumes it has a parent. In this new code, it allows it to be spawned without being assigned to a parent. It's an intentional change to permit it or other hat-based items to be placed as entities or spawned freely.

Reply:
@ZenBre4ker please mark this as resolved if your question is answered and merge :)